### ISSUE TITLE
Introduce estimatedMergeBytes attribute in SegmentInfo

### ISSUE THREAD

**Issue by sarthakaggarwal97** [apache/lucene#13278]
### Description

In OpenSearch, we are proposing the idea of [Hybrid Compression](https://github.com/opensearch-project/OpenSearch/issues/13110). With this, we are planning to take the compression off the write path and instead compress the data upon the merges above a certain size (of the merged segment) threshold. This would free up compute during active indexing with a trade off on disk. 

We are observing **5-14%** latency improvements in append-only cases and **5.5-16%** latency improvements with the update workload using the [nyc_taxis](https://github.com/opensearch-project/opensearch-benchmark-workloads/tree/main/nyc_taxis) dataset.

In order to get the estimated size of the merged segment, it would be useful to add ```estimatedMergeBytes``` in the SegmentInfo during [mergeInit](https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/index/IndexWriter.java#L4937). 

I am proposing to make this [change](https://github.com/sarthakaggarwal97/lucene/commit/9eb2faf4404b01aefc1b5f62bd3bdebbec3999b4), and would really appreciate some feedback from the community on this.

**Comment by jpountz** [comment]
Have you tried to take advantage of `IOContext` for this? It already exposes this information.

**Comment by sarthakaggarwal97** [comment]
I have not tried this. Thanks for point this out @jpountz 
Will close this issue!

**Comment by jpountz** [comment]
You can check out `NRTCachingDirectory` for an example of something that takes advantage of this.

**Comment by sarthakaggarwal97** [comment]
yeah, this is helpful. I was also thinking of setting dynamic compression modes during segment flushes, there FlushInfo could be handy. thank you!
